### ISSUE TITLE
A bidirectional relationship `parent.children` becomes `hasLoaded: true` when a child that references the parent is pushed to the store

### ISSUE THREAD

**Issue by lolmaus** [emberjs/data#5211]
## Affected versions

Ember Data 2.14+. This includes current stable Ember Data 2.16.1.

## Steps to reproduce

Given two models:

```js
// parent
export default Model.extend({
  children: hasMany('child')
});
```

```js
// child
export default Model.extend({
  parent: belongsTo('parent')
});
```

Push two records into the store:

```js
    const parent = store.push({
      data: {
        id: '1',
        type: 'parent',
        
        relationships: {
          children: {
            links: {
              "self": "https://example.com/parents/1/relationships/children",
              "related": "https://example.com/parents/1/children"
            }
          }
        }
      }
    })

    const child = store.push({
      data: {
        id: '1',
        type: 'child',
        relationships: {
          parent: {
            data: {
              id: '1',
              type: 'parent'
            }
          }
        }
      }
    })
```

## Expected behavior (Ember Data 2.13 and below work correctly)

The `parent.children` relationship is not considered loaded:

```js
parent.hasMany('children').hasManyRelationship.hasLoaded // => false
```

When you do `parent.get('children')`, a network request is made.

## Actual behavior (Ember Data 2.14 and above)

The `parent.children` relationship is considered loaded:

```js
parent.hasMany('children').hasManyRelationship.hasLoaded // => true
```

When you do `parent.get('children')`, a network request is *not* made, which breaks our app in many ways.

## Reproduction

https://ember-twiddle.com/3331c5dfc361920818fa3325842f5b3e?openFiles=controllers.application.js%2C (see console output, then try switching to `2.12.0` in `twiddle.json`)

## Credit

This bug has been discovered by @vvainio.

**Comment by lolmaus** [comment]
This happens not only after a push. You can `findRecord` or `query` for a child, and if it references a parent, the parent will start thinking that it has only one child.

**Comment by lolmaus** [comment]
This one seems to be either related or duplicate: #5209.

**Comment by vvainio** [comment]
Here's another Twiddle demonstrating this issue. In our use-case, we are fetching some data in the route's model hook, and then side loading the rest. This pattern has worked at least up until `2.12.2`.

https://ember-twiddle.com/45d37cc9b2525b32f3cd6dbda1c96ecf?fileTreeShown=false&numColumns=2&openFiles=routes.application.js%2Ctemplates.application.hbs

**Comment by BryanCrotaz** [comment]
#5209 isn't the same issue - it only happens when inverse on one end is null

**Comment by BryanCrotaz** [comment]
I have a failing test for this one - I'll try and fix tomorrow.

**Comment by BryanCrotaz** [comment]
Fixed in PR #5256 
@igorT @stefanpenner Can we get this reviewed and merged quickly? It's a major bug that changes client side data if using links.

**Comment by bmac** [comment]
I believe this has been resolved by https://github.com/emberjs/data/pull/5230. Feel free to reopen if it is still an issue.
